File: Online Stock Span.cpp
class StockSpanner {
    
public:
    
    vector<pair<int,int>>ss;
    StockSpanner() {
      
      ss.push_back({INT_MAX,1})  ;  
    }
    
    int next(int price) {
     
        int span=1;
        for(int i=ss.size()-1;i>0;)
        {
            if(ss[i].first<=price)
            {
                span+=ss[i].second;
                i-=ss[i].second;
            }
            else
            {
                break;
            }
            
        }
        ss.push_back({price,span});
        return span;
        
            
    }
};

/**
 * Your StockSpanner object will be instantiated and called as such:
 * StockSpanner* obj = new StockSpanner();
 * int param_1 = obj->next(price);
 */

File: Golang Solutions/Online Stock Span.go
type StockSpanner struct {
   ss1 []int
   ss2 []int
}


func Constructor() StockSpanner {
    var s StockSpanner
    s.ss1=append(s.ss1,9999999)
    s.ss2=append(s.ss2,1)
    return s
}


func (this *StockSpanner) Next(price int) int {
    span:=1
    for i:=len(this.ss1)-1;i>0;{
        if this.ss1[i]<=price{
            span+=this.ss2[i];
            i-=this.ss2[i]
        }else{
            break;
        }
        
    }
    this.ss1=append(this.ss1,price)
    this.ss2=append(this.ss2,span)
    return span
}


/**
 * Your StockSpanner object will be instantiated and called as such:
 * obj := Constructor();
 * param_1 := obj.Next(price);
 */